### PR TITLE
feat(system_log): move system log to its own plugin

### PR DESCRIPTION
**Pull request by hypeJunction**
Logging API has been moved to its own system_log plugin. logbrowser and logrotate
plugins have been combined into system_log plugin.

### CODE REVIEW

File: mod/system_log/lib/deprecated.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
+<?php
```
Comment (jdalsem) [inline mod/system_log/lib/deprecated.php]: do we need this deprecated? without the plugin active deprecation notices will never be triggered...

**Comment by jdalsem** [comment]
fixes #10811

File: mod/system_log/lib/deprecated.php (hunk 1 above)
Comment (hypeJunction) [reply mod/system_log/lib/deprecated.php]: Are you suggesting we just remove them and put into docs?

Comment (jdalsem) [reply mod/system_log/lib/deprecated.php]: yeah... i think so... don't think these functions will be called a lot directly

**Comment by hypeJunction** [comment]
Could someone restart the build. Weird failure

**Comment by jeabakker** [comment]
> Could someone restart the build.

done

**Comment by hypeJunction** [comment]
This is green

**Pull request merged**